experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -276,7 +296,15 @@ export interface MapNode<TSchema extends MapSchema> extends TreeNode {
 	// TODO: Add `set` method when FieldKind provides a setter (and derive the type from it).
 	// set(key: FieldKey, content: FlexibleFieldContent<TSchema["mapFields"]>): void;
 
-	[Symbol.iterator](): Iterator<TypedField<TSchema["mapFields"]>>;
+	/**
+	 * Iterate through all fields in the map.
+	 *
+	 * @remarks
+	 * The map may not be mutated during iteration.
+	 */
+	boxedIterator(): IterableIterator<TypedField<TSchema["mapFields"]>>;
+
+	[Symbol.iterator](): IterableIterator<UnboxField<TSchema["mapFields"], "notEmpty">>;
```

Comment:
This version should get a comment (and the other can inherit one: but if you are going to overide the comment from the base type, might want to inherit doc, then add a remark about having a non boxed one via Symbot.iterator.